alf/utils/tensor_utils.py

```diff
@@ -142,6 +142,8 @@ def global_norm(tensors):
     """
     assert alf.nest.is_nested(tensors), "tensors must be a nest!"
     tensors = alf.nest.flatten(tensors)
+    if not tensors:
+        return torch.zeros((), dtype=torch.float32)
```

Comment:
If not check here, torch.sqrt will complain the arg is int for an empty list.

Reply:
This function can potentially be used for other purposes so it's better to return a tensor.